src/script_opt/ScriptOpt.cc

```diff
@@ -253,15 +249,15 @@ static void optimize_func(ScriptFunc* f, std::shared_ptr<ProfileFunc> pf, ScopeP
 
 	if ( analysis_options.gen_ZAM_code )
 		{
-		ZAM = new ZAMCompiler(f, pf, scope, new_body, ud, rc);
```

Comment:
Were we never deleting this before?

Reply:
Indeed, we weren't. I'm pretty sure this was because doing so led to crashes because of the lack of rigor in tracking AST elements, a broader problem than just ZAM (though it was the bulk of it).